pkg/proxy/nftables/proxier.go

```diff
@@ -1522,6 +1522,13 @@ func (proxier *Proxier) syncProxyRules() {
 	// short amount of time later that the chain is now unreferenced. So we flush them
 	// now, and record the time that they become stale in staleChains so they can be
 	// deleted later.
+
+	// If a previously stale chain becomes active now, delete it from the staleChains.
+	for chain := range proxier.staleChains {
+		if activeChains.Has(chain) {
+			delete(proxier.staleChains, chain)
+		}
+	}
```

Comment:
hm... I think it would be better to modify the code a few lines down; we skip calling `tx.Flush()` if the chain is in `activeChains`, but we ought to actually be removing it from `staleChains`.

Reply:
got it, updated